c_wrapper/lua-mongoc-collection.c

```diff
@@ -731,25 +732,35 @@ lua_mongo_collection_drop_index(lua_State *L)
     bool ret = false;
     bool throw_error = false;
 
+    bson_t keys = BSON_INITIALIZER;
     collection_t *collection;
     bson_error_t error;
 
-    const char *index_name;
+    char *index_name;
     int absolute_luaBSONObjects_index = 2;
     int index_name_index = 3;
 
-    index_name = luaL_checkstring(L, index_name_index);
+    if(lua_isstring(L, index_name_index)){
+        index_name = luaL_checkstring(L, index_name_index);
+    }else if(lua_istable(L, index_name_index)){
+        throw_error = !(lua_table_to_bson(L, &keys, index_name_index, false, absolute_luaBSONObjects_index, &error ));
+        if(throw_error){
+            goto DONE;
+        }
+        index_name = mongoc_collection_keys_to_index_string(&keys);
+    }
     collection = (collection_t *) luaL_checkudata(L, 1, "lua_mongoc_collection");
 
     ret = mongoc_collection_drop_index(collection->c_collection, index_name, &error);
-    lua_pushboolean(L, ret);
+    lua_pushnil(L);
```

Comment:
You don't have to push nil here. You just `return 0;` instead of `return 1;`. An example in lua:

```
> function no_return() end
> ret_val = no_return()
> print(ret_val)
nil
```